neutron-physics/model.py

```diff
@@ -0,0 +1,387 @@
+
+        # Convert cross section data
+        data = openmc.data.IncidentNeutron.from_ace(table, 'mcnp')
+        self._temperature = data.kTs[0] / openmc.data.K_BOLTZMANN
+
+        # Export HDF5 file
+        os.makedirs('openmc', exist_ok=True)
+        h5_file = Path('openmc') / f'{data.name}.h5'
+        data.export_to_hdf5(h5_file, 'w')
+
+        # Register with library and write cross_sections.xml
+        data_lib = openmc.data.DataLibrary()
+        data_lib.register_file(h5_file)
+        data_lib.export_to_xml(Path('openmc') / 'cross_sections.xml')
+
+    def _make_openmc_input(self):
+        """Generate the OpenMC input XML
+
+        """
+        # Directory from which openmc is run
+        os.makedirs('openmc', exist_ok=True)
+        
+        # Define material
+        mat = openmc.Material()
+        mat.add_nuclide(self.nuclide, 1.0)
+        mat.set_density('g/cm3', self.density)
+        materials = openmc.Materials([mat])
+        if self.library is not None:
+            xs_path = (Path('openmc') / 'cross_sections.xml').resolve()
+            materials.cross_sections = str(xs_path)
+        materials.export_to_xml(Path('openmc') / 'materials.xml')
+
+        # Set up geometry
+        sphere = openmc.Sphere(boundary_type='reflective', R=1.e9)
+        cell = openmc.Cell(fill=materials, region=-sphere)
+        geometry = openmc.Geometry([cell])
+        geometry.export_to_xml(Path('openmc') / 'geometry.xml')
+
+        # Define source
+        source = openmc.Source()
+        source.space = openmc.stats.Point((0,0,0))
+        source.angle = openmc.stats.Isotropic()
+        source.energy = openmc.stats.Discrete([self.energy], [1.])
+
+        # Settings
+        settings = openmc.Settings()
+        if self._temperature is not None:
+            settings.temperature = {'default': self._temperature}
+        settings.source = source
+        settings.particles = self.particles
+        settings.run_mode = 'fixed source'
+        settings.batches = 1
+        settings.create_fission_neutrons = False
+        settings.export_to_xml(Path('openmc') / 'settings.xml')
+ 
+        # Define tallies
+        energy_bins = np.logspace(-5, np.log10(self.energy), 500)
+        energy_filter = openmc.EnergyFilter(energy_bins)
+        tally = openmc.Tally(name='neutron flux')
+        tally.filters = [energy_filter]
+        tally.scores = ['flux']
+        tallies = openmc.Tallies([tally])
+        tallies.export_to_xml(Path('openmc') / 'tallies.xml')
+
+    def _make_mcnp_input(self):
+        """Generate the MCNP input file
+
+        """
+        # Directory from which MCNP will be run
+        os.makedirs('mcnp', exist_ok=True)
+
+        # Create the problem description
+        lines = ['Point source in infinite geometry']
+ 
+        # Create the cell cards: material 1 inside sphere, void outside
+        lines.append('c --- Cell cards ---')
+        lines.append(f'1 1 -{self.density} -1 imp:n=1')
```

Comment:
Apparently you also need `tmp=value` here where value is kT in units of MeV. MCNP uses a free gas scattering treatment that defaults to T=293.6K and applies that in the absence of any temperature specification. The following should do:
```Python
kT = self._temperature * openmc.data.K_BOLTZMANN * 1e-6
...
lines.append(f'1 1 -{self.density} -1 imp:n=1 tmp={kT}')
```

Reply:
Ah, thanks for catching this.

Reply:
Yeah, I looked at some results for 600 K and they looked way off. Knew something had to be wrong!